packages/image_picker/image_picker_android/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerDelegate.java

```diff
@@ -253,10 +263,18 @@ public void chooseVideoFromGallery(MethodCall methodCall, MethodChannel.Result r
   }
 
   private void launchPickVideoFromGalleryIntent() {
-    Intent pickVideoIntent = new Intent(Intent.ACTION_GET_CONTENT);
+    boolean isPhotoPickerAvailable = ImagePickerUtils.isPhotoPickerAvailable();
+
+    Intent pickVideoIntent =
+        new Intent(
+            isPhotoPickerAvailable ? MediaStore.ACTION_PICK_IMAGES : Intent.ACTION_GET_CONTENT);
     pickVideoIntent.setType("video/*");
+    int requestCode =
+        isPhotoPickerAvailable
+            ? REQUEST_CODE_CHOOSE_VIDEO_FROM_GALLERY_USING_PHOTO_PICKER
+            : REQUEST_CODE_CHOOSE_VIDEO_FROM_GALLERY;
```

Comment:
No need for these to be different from each other. What is needed is just that they're different from anything else (any request codes anywhere in the app — not sure what the practice is among Flutter plugins for avoiding accidental conflicts between them), and that this plugin's `onActivityResult` handles them.

Reply:
Ok perfect! 

I initially made these different so the tests could confirm that the correct intent tag was being used based on the given Android SDK.

Reply:
(continued post-migration at https://github.com/flutter/packages/pull/3267/files#r1114883049 )